pkg/kstatus/observe/collector/collector.go

```diff
@@ -0,0 +1,111 @@
+package collector
+
+import (
+	"sort"
+	"sync"
+
+	"sigs.k8s.io/cli-utils/pkg/kstatus/observe/event"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/status"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/wait"
+)
+
+func NewObservedStatusCollector(identifiers []wait.ResourceIdentifier) *ObservedStatusCollector {
+	observations := make(map[wait.ResourceIdentifier]*event.ObservedResource)
+	for _, id := range identifiers {
+		observations[id] = &event.ObservedResource{
+			Identifier: id,
+			Status:     status.UnknownStatus,
+		}
+	}
+	return &ObservedStatusCollector{
+		aggregateStatus: status.UnknownStatus,
+		observations:    observations,
+	}
+}
+
+// ObservedStatusCollector is for use by clients of the observe library and provides
+// a way to keep track of the latest status/state for all the observed resources. The collector
+// is set up to listen to the eventChannel and keep the latest event for each resource. It also
```

Comment:
it would be helpful to see a doc.go explaining the big picture of 'observe' package usage,